diff --git a/Makefile b/Makefile
index a49870d3d1..1d10ae8da8 100644
--- a/Makefile
+++ b/Makefile
@@ -420,7 +420,7 @@ others: $(common-objpfx)testrun.sh $(common-objpfx)debugglibc.sh

 subdir-stubs := $(foreach dir,$(subdirs),$(common-objpfx)$(dir)/stubs)

-ifndef abi-variants
+ifdef abi-variants
 installed-stubs = $(inst_includedir)/gnu/stubs.h
 else
 installed-stubs = $(inst_includedir)/gnu/stubs-$(default-abi).h

