diff --git a/config/arch/k1.in b/config/arch/k1.in
new file mode 100644
index 00000000..3f1fd5c8
--- /dev/null
+++ b/config/arch/k1.in
@@ -0,0 +1,12 @@
+# K1 specific config options
+# This is deprecated and only here to allow build of previously published compiler on compiler-explorer.
+
+## no-package
+## select ARCH_SUPPORTS_32
+## select ARCH_SUPPORTS_64
+## select ARCH_DEFAULT_64
+## select ARCH_REQUIRES_MULTILIB
+## select ARCH_USE_MMU
+##
+## help The KVX MPPA Processor architectures, as defined by:
+## help     http://www.kalrayinc.com
diff --git a/config/arch/kvx.in b/config/arch/kvx.in
new file mode 100644
index 00000000..9e1cf74a
--- /dev/null
+++ b/config/arch/kvx.in
@@ -0,0 +1,11 @@
+# KVX specific config options
+
+## no-package
+## select ARCH_SUPPORTS_32
+## select ARCH_SUPPORTS_64
+## select ARCH_DEFAULT_64
+## select ARCH_REQUIRES_MULTILIB
+## select ARCH_USE_MMU
+##
+## help The KVX MPPA Processor architectures, as defined by:
+## help     http://www.kalrayinc.com
diff --git a/scripts/config.sub b/scripts/config.sub
index 3b4c7624..1907f4a1 100755
--- a/scripts/config.sub
+++ b/scripts/config.sub
@@ -1182,6 +1182,8 @@ case $cpu-$vendor in
 			| i370 | i*86 | i860 | i960 | ia16 | ia64 \
 			| ip2k | iq2000 \
 			| k1om \
+			| kvx \
+			| k1 \
 			| le32 | le64 \
 			| lm32 \
 			| m32c | m32r | m32rle \
